src/api/schema/components/mod.rs

```diff
@@ -68,7 +69,7 @@ impl filter::CustomFilter<Component> for ComponentsFilter {
         filter_check!(
             self.component_id.as_ref().map(|f| f
                 .iter()
-                .all(|f| f.filter_value(component.get_component_id()))),
+                .all(|f| f.filter_value(&component.get_component_id()))),
```

Comment:
The borrow here can be dropped if the suggestion to return `&str` above is used.